app/module/drivers/kscan/kscan_composite.c

```diff
@@ -62,6 +69,12 @@ static int kscan_composite_disable_callback(const struct device *dev) {
     for (int i = 0; i < cfg->children_len; i++) {
         const struct kscan_composite_child_config *child_cfg = &cfg->children[i];
 
+#if IS_ENABLED(CONFIG_PM_DEVICE_RUNTIME) || IS_ENABLED(CONFIG_PM_DEVICE)
+        if (pm_device_wakeup_is_enabled(dev) && pm_device_wakeup_is_enabled(child_cfg->child)) {
+            continue;
+        }
```

Comment:
I would expect this to *disable* wakeup on the child device when the composite is getting disabled.